conans/client/profile_loader.py

```diff
@@ -0,0 +1,245 @@
+import os
+from collections import OrderedDict
+
+from conans.errors import ConanException
+from conans.model.env_info import EnvValues, unquote
+from conans.model.info import ConanInfo
+from conans.model.options import OptionsValues
+from conans.model.profile import Profile
+from conans.model.ref import ConanFileReference
+from conans.model.scope import Scopes
+from conans.paths import CONANINFO
+from conans.util.config_parser import ConfigParser
+from conans.util.files import load, mkdir
+
+
+class ProfileParser(object):
+
+    def __init__(self, text):
+        self.text = text
+
+    @property
+    def _defs_block_text(self):
+        """Returns the text for #includes and vars declaration"""
+        ret_lines = []
+        for line in self.text.splitlines():
+            if not line or line.strip().startswith("#"):
+                continue
+            elif line.strip().startswith("["):
+                return "\n".join(ret_lines)
+            else:
+                ret_lines.append(line)
+        return "\n".join(ret_lines)
```

Comment:
This conversion to text again is confusing, then parsing them again line by line for includes and vars.